modules/consentManagement.js

```diff
@@ -0,0 +1,171 @@
+  if (consentId) {
+    applyConsent(adUnits, consentId);
+    return nextFn.apply(context, args);
+  }
+
+  if (!Object.keys(cmpCallMap).includes(userCMP)) {
+    utils.logWarn(`CMP framework (${userCMP}) is not a supported framework.  Aborting consentManagement module and resuming auction.`);
+    return nextFn.apply(context, args);
+  }
+  cmpCallMap[userCMP].call();
+}
+
+// after we have grabbed ideal ID from CMP, apply the data to adUnits object and finish up the module
+function postLookup(consentString) {
+  if (cmpActive) {
+    if (typeof consentString !== 'string' || consentString === '') {
+      exitFailedCMP(`CMP returned unexpected value during lookup process; returned value was (${consentString}).`);
+    } else {
+      clearTimeout(timer);
+      consentId = consentString;
+      applyConsent(adUnits, consentString);
+      nextFn.apply(context, args);
+    }
+  }
+}
+
+function cmpTimedOut() {
+  if (cmpActive) {
+    exitFailedCMP('CMP workflow exceeded timeout threshold.');
+  }
+}
+
+function exitFailedCMP(message) {
+  cmpActive = false;
+  if (lookupFailureChoice === 'proceed') {
+    utils.logWarn(message + ' Resuming auction without consent data as per consentManagement config.');
+    applyConsent(adUnits, undefined);
+    nextFn.apply(context, args);
+  } else {
+    utils.logError(message + ' Canceling auction as per consentManagement config.');
+  }
+}
+
+// assuming we have valid consent ID, apply to adUnits object
+function applyConsent(adUnits, consentString) {
+  adUnits.forEach(adUnit => {
+    adUnit['gdprConsent'] = {
+      consentString: consentString,
+      consentRequired: true
+    };
+  });
+}
+
+/**
+ * Simply resets the module's consentId variable back to its default value.
+*/
+export function resetConsentId() {
+  consentId = '';
+}
+
+/**
+ * A configuration function that initializes some module variables, as well as add a hook into the requestBids function
+ * @param {object} config consentManagement module config settings; cmp (string), waitForConsentTimeout (int), lookUpFailureResolution (string)
+ */
+export function setConfig(config) {
+  if (typeof config.cmp === 'string') {
+    userCMP = config.cmp;
+  } else {
+    userCMP = DEFAULT_CMP;
+    utils.logInfo(`consentManagement config did not specify cmp.  Using system default setting (${userCMP}).`);
+  }
+
+  if (typeof config.waitForConsentTimeout === 'number') {
+    consentTimeout = config.waitForConsentTimeout;
+  } else {
+    consentTimeout = DEFAULT_CONSENT_TIMEOUT;
+    utils.logInfo(`consentManagement config did not specify waitForConsentTimeout.  Using system default setting (${consentTimeout}).`);
+  }
+
+  if (typeof config.lookUpFailureResolution === 'string') {
+    if (config.lookUpFailureResolution === 'proceed' || config.lookUpFailureResolution === 'cancel') {
+      lookupFailureChoice = config.lookUpFailureResolution;
+    } else {
+      lookupFailureChoice = DEFAULT_FAILURE_CHOICE;
```

Comment:
Is there any reason this config option shouldn't be reworked as a boolean?  It seems like right now it's just `'proceed'` or `'cancel'`.  true/false.  Is there more options I'm missing?

Reply:
At the moment, there are just these two conditions.  A boolean could work; what do you think of  the name `allowAuctionWithoutConsent`?  `true` would equate to previous `proceed` and `false` to `cancel`.